service/src/tests/mocks/parentchain_api_mock.rs

```diff
@@ -0,0 +1,72 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use itp_api_client_extensions::{ApiResult, ChainApi, StorageProof};
+use itp_test::builders::{
+	parentchain_block_builder::ParentchainBlockBuilder,
+	parentchain_header_builder::ParentchainHeaderBuilder,
+};
+use itp_types::{SignedBlock, H256};
+use sp_finality_grandpa::AuthorityList;
+use std::cmp::min;
+
+pub struct ParentchainApiMock {
+	parentchain: Vec<SignedBlock>,
+}
+
+impl ParentchainApiMock {
+	pub(crate) fn new() -> Self {
+		ParentchainApiMock { parentchain: Vec::new() }
+	}
+
+	// can this be combined with the constructor? Does it make sense?
+	pub fn create_mock_parentchain(&mut self, len: u32) {
```

Comment:
Usually what you would have in the mocks, is something along the lines of a builder pattern. So here you crate some default blocks. Then you would have a build method:
```
	pub fn with_default_blocks(mut self, number_of_blocks: u32) -> Self {
		for n in 1..=number_of_blocks {
			let header = ParentchainHeaderBuilder::default().with_number(n).build();
			let block = ParentchainBlockBuilder::default().with_header(header).build_signed();
			self.parentchain.push(block);
		}
		self
	}
```
(I think the entire vector generation can also be done in a more functional style, without using a for loop with pushes, but I'll have to quickly check how that is exactly done in rust)

Then you use it like this:
```
let parentchain_api_mock = ParentchainApiMock::new().with_default_blocks(6);
```